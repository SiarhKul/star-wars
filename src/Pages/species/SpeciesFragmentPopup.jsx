import React from "react";
import { useSelector } from "react-redux";
import { PopupListItem } from "../../components/popup/PopupListItem";

export const SpeciesFragmentPopup = () => {
	const {
		name,
		classification,
		designation,
		skin_colors,
		hair_colors,
		eye_colors,
		average_lifespan,
		language,
		people,
		films,
	} = useSelector(state => state.dataFromServer.clickedCard);

	console.log(
		name,
		classification,
		designation,
		skin_colors,
		hair_colors,
		eye_colors,
		average_lifespan,
		language,
		people,
		films
	);

	return (
		<>
			<div className="name-wrapper">
				<span className="popup__name">{name}</span>
			</div>
			<div className="description-wrapper">
				<div className="description-person">
					<div className="description-appearance">
						<p className="description__heading">Appearance</p>
						<ul className="description-appearance__item">
							<li>Classification:{classification}</li>
							<li>Designation:{designation}</li>
							<li>Skin colors:{skin_colors}</li>
							<li>Hair colors:{hair_colors}</li>
							<li>Eye colors:{eye_colors}</li>
							<li>Average_lifespan:{average_lifespan}</li>
							<li>Language:{language}</li>
						</ul>
					</div>
				</div>

				<div className="description-resources">
					<div className="resources">
						<div className="resources-heading">Residents:</div>
						<ul className="resources-list">
							<PopupListItem urls={people} name="name" path="/" />
						</ul>
					</div>

					<div className="resources">
						<div className="resources-heading">Films:</div>
						<ul className="resources-list">
							<PopupListItem urls={films} name="title" path="films" />
						</ul>
					</div>
				</div>
			</div>
		</>
	);
};
