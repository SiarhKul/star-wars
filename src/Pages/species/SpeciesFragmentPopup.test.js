import React from "react";
import { SpeciesFragmentPopup } from "./SpeciesFragmentPopup";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => {
			return {
				location: {
					state: {
						name: "Rodian",
						classification: "sentient",
						designation: "reptilian",
						skin_colors: "green, blue",
						hair_colors: "n/a",
						eye_colors: "black",
						average_lifespan: "unknown",
						people: ["https://swapi.dev/api/people/15/"],
						films: ["https://swapi.dev/api/films/1/"],
					},
				},
			};
		}),
	};
});

describe("Test <SpeciesFragmentPopup/>", () => {
	const speciesFragmentPopup = () => shallow(<SpeciesFragmentPopup />);
	let component;

	beforeEach(() => {
		component = speciesFragmentPopup();
	});

	it("render snapshot <SpeciesFragmentPopup/> ", () => {
		expect(component).toMatchSnapshot();
	});

	it("should  <SpeciesFragmentPopup/>  has required fields", () => {
		expect(component.text().includes("Appearance")).toBeTruthy();
		expect(component.text().includes("Designation:")).toBeTruthy();
		expect(component.text().includes("Skin colors:")).toBeTruthy();
		expect(component.text().includes("Hair colors:")).toBeTruthy();
		expect(component.text().includes("Eye colors:")).toBeTruthy();
		expect(component.text().includes("Language:")).toBeTruthy();
		expect(component.text().includes("Residents:")).toBeTruthy();
		expect(component.text().includes("Films:")).toBeTruthy();
	});
});
