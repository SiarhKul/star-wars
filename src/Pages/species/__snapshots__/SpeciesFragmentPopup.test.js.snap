// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SpeciesFragmentPopup/> render snapshot <SpeciesFragmentPopup/>  1`] = `
<Fragment>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    >
      Rodian
    </span>
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Classification:
            sentient
          </li>
          <li>
            Designation:
            reptilian
          </li>
          <li>
            Skin colors:
            green, blue
          </li>
          <li>
            Hair colors:
            n/a
          </li>
          <li>
            Eye colors:
            black
          </li>
          <li>
            Average lifespan:
            unknown
          </li>
          <li>
            Language:
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Residents:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="people"
            urls={
              Array [
                "https://swapi.dev/api/people/15/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Films:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="title"
            path="films"
            urls={
              Array [
                "https://swapi.dev/api/films/1/",
              ]
            }
          />
        </ul>
      </div>
    </div>
  </div>
</Fragment>
`;
